cro-clib/src/jsonrpc.rs

```diff
@@ -108,11 +110,41 @@ fn make_syncer_config(
     ))
 }
 
+use std::sync::Arc;
+use std::sync::Mutex;
+
+#[derive(Clone)]
+struct CBindingData {
+    progress_callback: ProgressCallback,
+    user_data: u64,
```

Comment:
probably add some safety warning about the unsafe cast (*void -> u64) that this should never be accessed from within rust and should be only be passed back to the progress callback

Reply:
ok